src/unittests/maths.spec.ts

```diff
@@ -27,24 +27,48 @@ import {
   linearRange,
   nextAfterF32,
   oneULP,
-  withinULP,
 } from '../webgpu/util/math.js';
 
 import { UnitTest } from './unit_test.js';
 
 export const g = makeTestGroup(UnitTest);
 
 /**
- * @returns true if arrays are equal within 1ULP, doing element-wise comparison as needed, and considering NaNs to be equal.
+ * Utility wrapper around oneULP to test if a value is within 1 ULP(x)
  *
- * Depends on the correctness of withinULP, which is tested in this file.
+ * @param got number to test
+ * @param expected number to be within 1 ULP of
+ * @param flush should oneULP FTZ
+ * @returns if got is within 1 ULP of expected
+ */
+function withinOneULP(got: number, expected: number, flush: boolean): boolean {
+  const ulp = oneULP(expected, flush);
+  return got >= expected - ulp && got <= expected + ulp;
+}
+
+/**
+ * @returns true if arrays are equal within 1ULP, doing element-wise comparison
+ *               as needed, and considering NaNs to be equal.
+ *
+ * Depends on the correctness of oneULP, which is tested in this file.
+ **
+ * @param got array of numbers to compare for equality
+ * @param expect array of numbers to compare against
+ * @param flush should different subnormals be considered the same, i.e. should
+ *              FTZ occur during comparison
  **/
-function compareArrayOfNumbers(got: Array<number>, expect: Array<number>): boolean {
+function compareArrayOfNumbers(
+  got: Array<number>,
+  expect: Array<number>,
+  flush: boolean = true
```

Comment:
I'd suggest using a  `type FlushMode = 'flush' | 'no-flush';` over boolean, for callsite readability's sake.

Reply:
Done